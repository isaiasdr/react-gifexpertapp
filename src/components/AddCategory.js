import React, { useState } from 'react';
import propTypes from 'prop-types';

export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue( e.target.value );
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length > 2) {
            setCategories( categories => { 
                if(!categories.find( categorie => categorie === inputValue))    
                    return [ inputValue, ...categories ];

                else
                    return [...categories];
            });

            setInputValue( '' );
        }
    }

    return (
        <form onSubmit = { handleSubmit }>
            <input 
                type="text"
                value = { inputValue }
                onChange = { handleInputChange }
            />
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: propTypes.func.isRequired
};