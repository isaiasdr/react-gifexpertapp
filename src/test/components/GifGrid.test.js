import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs'); 

describe('Test GifGrid', () => {
    const value = 'test'; 

    test('should show GifGrid correctly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow( <GifGrid category={value} /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('should show images when load images useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost:3000/werever',
            title: 'Wenas'
        },
        {
            id: 'ABC123',
            url: 'https://localhost:3000/werever',
            title: 'Wenas'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow( <GifGrid category={value} /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);

        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });
});