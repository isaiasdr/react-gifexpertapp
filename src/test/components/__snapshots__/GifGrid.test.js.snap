// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGrid should show GifGrid correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__pulse"
  >
    test
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test GifGrid should show images when load images useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__pulse"
  >
    test
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Wenas"
      url="https://localhost:3000/werever"
    />
    <GifGridItem
      id="ABC123"
      key="ABC123"
      title="Wenas"
      url="https://localhost:3000/werever"
    />
  </div>
</Fragment>
`;
